lib/Doctrine/ODM/MongoDB/Mapping/Annotations/AbstractField.php

```diff
@@ -4,9 +4,7 @@
 
 namespace Doctrine\ODM\MongoDB\Mapping\Annotations;
 
-use Doctrine\Common\Annotations\Annotation;
-
-abstract class AbstractField extends Annotation
+abstract class AbstractField implements Annotation
 {
     /** @var string */
```

Comment:
```suggestion
    /** @var string|null */
```
It is not related to this PR, but it would be nice to fix these types while modifying these classes (it can be done in another PR).

Reply:
Adding type hints to constructor arguments made it easier to spot some errors - many tests started to fail when I added  type hint for `cascade` which accepts both `array` and `string` while phpdoc stated it's `array` only 😉 
After I finish work on this PR, I can do one more round to check for more inconsistencies in a separate PR.

Reply:
Its fixing doc bugs that are needed for this PR, from my pov this can be combined here, we did the same for the ORM patch